frontend/src/service/backendService.ts

```diff
@@ -1,16 +1,51 @@
-function getBackendUrl(): string {
+function backendUrl(): string {
 	const url = import.meta.env.VITE_BACKEND_URL;
 	if (!url) throw new Error('VITE_BACKEND_URL env variable not set');
 
 	return url.endsWith('/') ? url : `${url}/`;
 }
 
 function makeUrl(path: string): URL {
-	return new URL(path, getBackendUrl());
+	return new URL(path, backendUrl());
 }
 
-async function sendRequest(path: string, options: RequestInit) {
-	return fetch(makeUrl(path), { ...options, credentials: 'include' });
+type RequestOptions<T> = Pick<RequestInit, 'method' | 'signal'> & {
+	body?: T;
+};
+
+async function get(
```

Comment:
This is all starting to look like [axios](https://axios-http.com/)! But hey, our implementation here is really lightweight and focused.